modules/search/trello.py

```diff
@@ -0,0 +1,82 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+
+meta = {
+	'name': 'Trello Search',
+	'author': 'Vikas Kundu',
+	'version': '0.5',
+	'description': 'Search public trello boards for data and show the results.',
+	'sources': ('google', 'carrot2', 'bing', 'yippy', 'yahoo', 'millionshort', 'qwant', 'duckduckgo'),
+	'options': (
+		('query', None, True, 'Query string', '-q', 'store', str),
+		('limit', 1, False, 'Search limit(number of pages, default=1)', '-l', 'store', int),
+		('count', 50, False, 'Number of results per page(min=10, max=100, default=50)', '-c', 'store', int),
+		('thread', 2, False, 'The number of engine that run per round(default=2)', '-t', 'store', int),
+		('engine', 'google', False, 'Engine names for search(default=google)', '-e', 'store', str),
+	),
+    'examples': ('trello -q <QUERY> -l 15 --output',)
+}
+
+
+LINKS = []
+PAGES = ''
+
+def search(self, name, q, q_formats, limit, count):
+	global PAGES,LINKS
+	engine = getattr(self, name)
+	name = engine.__init__.__name__
+	q = f"{name}_q" if f"{name}_q" in q_formats else q_formats['default_q']
+	varnames = engine.__init__.__code__.co_varnames
+	if 'limit' in varnames and 'count' in varnames:
+		attr = engine(q, limit, count)
+	elif 'limit' in varnames:
+		attr = engine(q, limit)
+	else:
+		attr = engine(q)
+
+	attr.run_crawl()
+	LINKS += attr.links
+	PAGES += attr.pages
+
+def module_api(self):
+	query = self.options['query']
+	limit = self.options['limit']
+	count = self.options['count']
+	engines = self.options['engine'].split(',')
+	output = {'links': [], 'usernames': [], 'handles': []}
+	q_formats = {
+		'default_q': f"site:trello.com {query}",
+		'yippy_q': f'"trello.com" {query}',
+		'millionshort_q': f'site:trello.com "{query}"',
+		'qwant_q': f'site:trello.com {query}'
+	}
+
+	self.thread(search, self.options['thread'], engines, query, q_formats, limit, count, meta['sources'])
+	
+	output['links'] = list(self.reglib().filter(r"https?://([\w\-\.]+\.)?trello\.com/", list(set(LINKS))))
+	output['usernames'] = [ i for i in set(re.findall(r"\([\w]{3,}\)", PAGES)) if query in i  ]
```

Comment:
Check this regex:
`\(([\w]{3,})\)`

Reply:
@saeeddhqan  usernames are present on search engine pages in this format (see image: https://ibb.co/TKYjPV1) . So it has to be within curly braces. So \(.*\). Now that it is of type word so \([\w]\). Now to more refine the search, i filter such usernames which do no contain the search query. The {3,} part is to make sure at least it is of length 3 or more. I can remove this part but the rest seems fine to me.

Reply:
It's not good to show usernames like this: (@abc)\n(@cba)

Reply:
@saeeddhqan ok so should i remove the usernames and handles part  ?

Reply:
I mean, this was?

Reply:
There are three parts to the output
1. Links: Normal trello board links
2. usernames: The ones inside round braces (vikas123)
3. handles: Those with mentions on public trello boards i.e. @trello123, @something
So, if i am understanding it, I need to remove no. 2 and no. 3 part. Right?

Reply:
> It's not good to show usernames like this: (@abc)\n(@cba)
It didn't mean to remove the usernames. it means, isn't it better to show usernames like this: @abc\n@cba
shall we?

Reply:
Oh. Got it. Will make the output neat.

Reply:
I think this regex \(([\w]{3,})\) will find what we want.

Reply:
will this do? https://ibb.co/jgt4Z5T

Reply:
A

Reply:
Made the necessary changes https://github.com/saeeddhqan/Maryam/pull/100/commits/2a6e367a979564889a740319303763e6d986b9eb